### PR TITLE
feat: add infrastructure tabs and content

### LINKED ISSUE #2552: Display `ftl status` information in Infrastructure section
- [ ] Build navigation
- [ ] Implement component specific pages for 
  - [ ] Controllers
  - [ ] Runners
  - [ ] Deployments
  - [ ] Routes

### PR DESCRIPTION
**Pull request by wesbillman**
Fixes #2552 

Screenshot 2024-09-03 at 1 44 20 PM
Screenshot 2024-09-03 at 1 44 27 PM
Screenshot 2024-09-03 at 1 44 30 PM
Screenshot 2024-09-03 at 1 44 35 PM

### CODE REVIEW

File: frontend/console/src/components/List.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { classNames } from '../utils'
+
+type ListProps<T> = {
+  items: T[]
+  renderItem: (item: T) => React.ReactNode
+  onClick?: (item: T) => void
+  className?: string
+}
+
+export const List = <T,>({ items, renderItem, onClick, className }: ListProps<T>) => {
```
Comment (deniseli) [inline frontend/console/src/components/List.tsx:10]: Did you mean to remove this `,` or is that doing some fancy JS magic I haven't learned yet? :)

File: frontend/console/src/components/List.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <ul className={classNames('divide-y divide-gray-100 dark:divide-gray-700 overflow-hidden', className)}>
+      {items.map((item, index) => (
+        <li
+          key={index}
+          className={`relative flex justify-between gap-x-6 p-4 ${onClick ? 'cursor-pointer hover:bg-gray-50 dark:hover:bg-gray-700' : ''}`}
+          onClick={onClick ? () => onClick(item) : undefined}
```
Comment (deniseli) [inline frontend/console/src/components/List.tsx:17]: Just to confirm, if you use this component without an `onClick`, does passing `undefined` into this prop definitely not throw a runtime error? I wasn't sure if it was safe to pass `undefined` to a `fn` prop.

File: frontend/console/src/features/infrastructure/RoutesList.tsx [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+export const RoutesList = ({ routes }: { routes: StatusResponse_Route[] }) => {
+  return (
+    <List
+      items={routes}
+      renderItem={(route) => (
+        <div className='flex w-full'>
+          <div className='flex gap-x-4 items-center w-1/2'>
+            <div className='whitespace-nowrap'>
+              <div className='flex gap-x-2 items-center'>{route.module}</div>
+              <p className='mt-1 flex text-xs leading-5 text-gray-400 font-roboto-mono'>{route.endpoint}</p>
```
Comment (deniseli) [inline frontend/console/src/features/infrastructure/RoutesList.tsx:14]: nit: this shade of gray feels good for dark mode but a little eye-strainy for light mode. What do you think about bumping it up to 600?

File: frontend/console/src/components/StatusIndicator.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+type StatusIndicatorProps = {
+  state: 'success' | 'error' | 'idle'
+  text?: string
+}
+
+export const StatusIndicator: React.FC<StatusIndicatorProps> = ({ state, text }) => {
+  const backgrounds = {
+    idle: 'text-gray-500 bg-gray-100/10',
+    success: 'text-green-400 bg-green-400/10',
+    error: 'text-rose-400 bg-rose-400/10',
```
Comment (deniseli) [inline frontend/console/src/components/StatusIndicator.tsx:13]: I feel like these also look great on dark mode but feel a little light for light mode. What do you think?

**Review by deniseli — APPROVED** [review]

File: frontend/console/src/components/List.tsx (hunk 1 above)
Comment (wesbillman) [reply frontend/console/src/components/List.tsx:10]: I think the `,` is required for the component to parse correctly. Removing it causes all kinds of errors 😂 . I think it's just a react generic component feature.

File: frontend/.gitignore [hunk 5]
```diff
@@ -7,6 +7,9 @@ yarn-error.log*
 pnpm-debug.log*
 lerna-debug.log*
 
+playwright-report/
+test-results/
+
```
[the remark this answers is not in the document]
Comment (wesbillman) [reply frontend/.gitignore:12]: You bet! I think there are going to be other similar issues after the relocating of all these files.

File: frontend/console/src/components/List.tsx (hunk 2 above)
Comment (wesbillman) [reply frontend/console/src/components/List.tsx:17]: yeah should be ok. All the current stuff is using it without an `onClick`.

File: frontend/console/src/components/List.tsx (hunk 1 above)
Comment (deniseli) [reply frontend/console/src/components/List.tsx:10]: oh wow that's good to know!

File: frontend/console/src/components/StatusIndicator.tsx (hunk 4 above)
Comment (wesbillman) [reply frontend/console/src/components/StatusIndicator.tsx:13]: Good call! Updated.

File: frontend/console/src/features/infrastructure/RoutesList.tsx (hunk 3 above)
Comment (wesbillman) [reply frontend/console/src/features/infrastructure/RoutesList.tsx:14]: Bumped to `500` which feels pretty good too

**Comment by alecthomas** [comment]
GLORIOUS!

**Pull request merged**